diff -p -up squashfs3.3/dkms/inode.c.2627 squashfs3.3/dkms/inode.c
--- squashfs3.3/dkms/inode.c.2627	2008-09-02 13:16:49.000000000 +0200
+++ squashfs3.3/dkms/inode.c	2008-09-02 13:20:34.000000000 +0200
@@ -2301,7 +2301,9 @@ static void squashfs_destroy_inode(struc
 }
 
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 24)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 27)
+static void init_once(void *foo)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 24)
 static void init_once(struct kmem_cache *cachep, void *foo)
 #else
 static void init_once(void * foo, struct kmem_cache * cachep, unsigned long flags)
